<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Jacques van Helden" />

<meta name="date" content="2022-02-11" />

<title>Practical: analysis of ChIP-seq peaks</title>

<script src="chip-seq_analysis_files/header-attrs-2.11/header-attrs.js"></script>
<script src="chip-seq_analysis_files/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="chip-seq_analysis_files/bootstrap-3.3.5/css/cerulean.min.css" rel="stylesheet" />
<script src="chip-seq_analysis_files/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="chip-seq_analysis_files/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="chip-seq_analysis_files/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="chip-seq_analysis_files/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="chip-seq_analysis_files/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="chip-seq_analysis_files/tocify-1.9.1/jquery.tocify.js"></script>
<script src="chip-seq_analysis_files/navigation-1.1/tabsets.js"></script>
<script src="chip-seq_analysis_files/navigation-1.1/codefolding.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>


<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  background-color: #f8f8f8; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ef2929; } /* Alert */
code span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #c4a000; } /* Attribute */
code span.bn { color: #0000cf; } /* BaseN */
code span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4e9a06; } /* Char */
code span.cn { color: #000000; } /* Constant */
code span.co { color: #8f5902; font-style: italic; } /* Comment */
code span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code span.dt { color: #204a87; } /* DataType */
code span.dv { color: #0000cf; } /* DecVal */
code span.er { color: #a40000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cf; } /* Float */
code span.fu { color: #000000; } /* Function */
code span.im { } /* Import */
code span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code span.ot { color: #8f5902; } /* Other */
code span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code span.sc { color: #000000; } /* SpecialChar */
code span.ss { color: #4e9a06; } /* SpecialString */
code span.st { color: #4e9a06; } /* String */
code span.va { color: #000000; } /* Variable */
code span.vs { color: #4e9a06; } /* VerbatimString */
code span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */

.sourceCode .row {
  width: 100%;
}
.sourceCode {
  overflow-x: auto;
}
.code-folding-btn {
  margin-right: -30px;
}
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>







<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>



<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Practical: analysis of ChIP-seq peaks</h1>
<h3 class="subtitle">LCG_BEII 2019</h3>
<h4 class="author">Jacques van Helden</h4>
<h4 class="date">2022-02-11</h4>

</div>


<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>In this practical, we will combine various bioinformatics resources
to extract information from ChIP-seq experiments, and evaluate the
quality of the peaks.</p>
</div>
<div id="resources" class="section level2">
<h2>Resources</h2>
<table>
<colgroup>
<col width="39%" />
<col width="60%" />
</colgroup>
<thead>
<tr class="header">
<th>Resource Name</th>
<th>URL</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>ReMap</td>
<td>A database of ChIP-seq peaks (<a href="https://remap.univ-amu.fr/"
class="uri">http://remap.cisreg.eu/</a>)</td>
</tr>
<tr class="even">
<td>Jaspar</td>
<td>A database of eukaryote TF binding motifs, mainly built from
CHIP-seq peaks (<a href="http://jaspar.genereg.net/"
class="uri">http://jaspar.genereg.net/</a>)</td>
</tr>
<tr class="odd">
<td>Hocomoco</td>
<td>A database of Human TF binding motifs, mainly built from CHIP-seq
peaks (<a href="http://hocomoco11.autosome.ru/"
class="uri">http://hocomoco11.autosome.ru/</a>)</td>
</tr>
<tr class="even">
<td>RSAT</td>
<td>Regulatory Sequence Analysis Tools <a href="http://rsat.eu/"
class="uri">http://rsat.eu/</a></td>
</tr>
<tr class="odd">
<td>RSAT Metazoa</td>
<td><a href="http://metazoa.rsat.eu/"
class="uri">http://metazoa.rsat.eu/</a></td>
</tr>
<tr class="even">
<td>RSAT Teaching</td>
<td><a href="http://teaching.rsat.eu/"
class="uri">http://teaching.rsat.eu/</a></td>
</tr>
<tr class="odd">
<td>RSAT server at IFB</td>
<td><a href="http://rsat.france-bioinformatique.fr/rsat/"
class="uri">http://rsat.france-bioinformatique.fr/rsat/</a></td>
</tr>
<tr class="even">
<td>UCSC NGS file formats</td>
<td><a href="https://genome.ucsc.edu/FAQ/FAQformat.html"
class="uri">https://genome.ucsc.edu/FAQ/FAQformat.html</a></td>
</tr>
</tbody>
</table>
</div>
<div id="creating-a-workspace-for-this-practical"
class="section level2">
<h2>Creating a workspace for this practical</h2>
<ul>
<li>Create a new folder to store the results of this practical
(e.g. <code>~/LCG_BEII/chip-seq_practical/</code>)</li>
<li>Set this folder as your working direct</li>
</ul>
</div>
<div id="getting-peaks-from-remap" class="section level2">
<h2>Getting peaks from ReMap</h2>
<p>ReMap (<a href="https://remap.univ-amu.fr/"
class="uri">https://remap.univ-amu.fr/</a>) is a database of TF binding
regions based on an extensive re-analysis of published ChIP-seq data for
human transcription factors. ory (<code>setwd()</code>)</p>
<ul>
<li><p>Open a connection to <a
href="https://remap.univ-amu.fr/">ReMap</a></p></li>
<li><p>Select <em>Homo sapiens</em></p></li>
<li><p>Explore the interface</p></li>
<li><p>Choose a transcription factor and a tissue / cell type of
interest.</p>
<ul>
<li>For a same factor, you wil generally find several datasets resulting
from different studies, in different tissues (biotypes).</li>
<li>For your exercise, I recommend to select a dataset with a reasonable
number of peaks</li>
<li>For this tutorial I will use Sox2 in <a
href="https://remap.univ-amu.fr/biotype_page/glioma:9606">glioma</a>,
where ReMap peaks were identified from the original <a
href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE67282">GEO
dataset GSE67282</a></li>
</ul></li>
<li><p>Click on the <strong>Download icon</strong> tab, besides the
number of peaks. This downloads a compressed (gzipped) bed file
(tab-separated columns describing genomic features).</p></li>
<li><p>Check that the number of peaks in the downloaded file corresponds
to the indication on the ReMap page for your factor in the tissue / cell
type and in the study identifier (GSE or other) you selected.</p></li>
<li><p>Check the <em>name</em> column in the bed file to make sure that
the peaks all belong to the same tissue / cell type and study
identifier.</p></li>
<li><p>Compute</p>
<ul>
<li>the number of peaks</li>
<li>the sum of peak lengths</li>
<li>the mean peak length</li>
<li>the median peak length</li>
</ul></li>
</ul>
<p><strong>Beware:</strong> the bed format relies on a somewhat
confusing convention to specify the coordinates of genomic features.</p>
<ul>
<li>Coordinates are 0-based (the first position of a sequence is
referred to as 0).</li>
<li>The feature start is (as expected) the first position within the
feature, but the “end” is the first position <strong>after the
feature</strong>.</li>
<li>Consequently, the feature length can be computed as <span
class="math inline">\(l = end - start\)</span>, whereas with other
formats it should be <span class="math inline">\(l = end - start =
1\)</span>.</li>
</ul>
<p><strong>Tips:</strong></p>
<ul>
<li><p>The peak statistics can be computed with a spreadsheet (Excel,
LibreOffice Calc), R, or the Unix command line.</p></li>
<li><p>Click the “code” button to display a bash line to compute peak
statistics (number, sum of lengths and mean length) from a
<code>bed</code>file.</p></li>
</ul>
<div class="sourceCode" id="cb1"><pre
class="sourceCode bash"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="va">RESULT_DIR</span><span class="op">=</span>~/LCG_BEII/chip-seq_practical/SOX2_glioma_GSE67282</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="va">PEAKS</span><span class="op">=</span><span class="va">${RESULT_DIR}</span>/GSE67282.SOX2.glioma_stem.bed.gz</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">gunzip</span> <span class="at">-c</span> <span class="va">${PEAKS}</span> <span class="dt">\</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a>   <span class="kw">|</span> <span class="fu">awk</span> <span class="st">&#39;{len=$3 - $2; sum += len; n++; mean = sum / n ; print &quot;n=&quot;n&quot;\tsum=&quot;sum&quot;\tmean=&quot;mean}&#39;</span> <span class="dt">\</span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a>   <span class="kw">|</span> <span class="fu">tail</span> <span class="at">-n</span> 1</span></code></pre></div>
<ul>
<li>Click the “code” button to display a chunk of R code to compute peak
statistics.</li>
</ul>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>resultDir <span class="ot">&lt;-</span> <span class="st">&quot;~/LCG_BEII/chip-seq_practical/SOX2_glioma_GSE67282&quot;</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>peakBase <span class="ot">&lt;-</span> <span class="st">&quot;GSE67282.SOX2.glioma_stem.bed.gz&quot;</span></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>peakFile <span class="ot">&lt;-</span> <span class="fu">file.path</span>(resultDir, peakBase)</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a><span class="co">#peakFile &lt;- &quot;data/ReMap_GSE67282.SOX2.ESC_peaks.bed&quot;</span></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>peaks <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(peakFile, <span class="at">header =</span> <span class="cn">FALSE</span>)</span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>peaks <span class="ot">&lt;-</span> peaks[, <span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>] <span class="co"># Keep the fields used for this analysis</span></span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(peaks) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;chr&quot;</span>, <span class="st">&quot;start&quot;</span>, <span class="st">&quot;end&quot;</span>, <span class="st">&quot;name&quot;</span>)</span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a>peaks<span class="sc">$</span>len <span class="ot">&lt;-</span> peaks<span class="sc">$</span>end <span class="sc">-</span> peaks<span class="sc">$</span>start</span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(peaks) <span class="co"># print the first peaks</span></span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a>peak_stats <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a>  <span class="at">sum =</span> <span class="fu">sum</span>(peaks<span class="sc">$</span>len),</span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a>  <span class="at">mean =</span> <span class="fu">mean</span>(peaks<span class="sc">$</span>len),</span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a>  <span class="at">n =</span> <span class="fu">nrow</span>(peaks))</span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-17"><a href="#cb2-17" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(knitr)</span>
<span id="cb2-18"><a href="#cb2-18" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(peak_stats)</span></code></pre></div>
<p>In our example (Sox2 peaks in embryonic stem cells from the GEO
series GSE67282) we obtain:</p>
<ul>
<li>Number of peaks: <span class="math inline">\(n = 3,859\)</span></li>
<li>Sum of peak lengths: <span class="math inline">\(sum =
1,154,851\)</span></li>
<li>Mean peak lengths: <span class="math inline">\(mean =
299.3\)</span></li>
</ul>
</div>
<div id="getting-motifs-from-reference-databases"
class="section level2">
<h2>Getting motifs from reference databases</h2>
<ul>
<li><p>Open a connection to <a
href="http://jaspar.genereg.net/">Jaspar</a></p>
<ul>
<li><p>Explore the database functionalities</p></li>
<li><p>Find the transcription factor binding motifs for your TF of
interest.</p></li>
<li><p>Download the corresponding matrix in your working directory.</p>
<ul>
<li>Note: JASPAR proposes several matrix formats. All these formats are
supported yb RSAT, but for convenience you can choose the TRANSFAC
format.</li>
</ul></li>
<li><p>Save this transfac-formatted motif in a text file named
<code>[YOUR_FACTOR_NAME]_reference-motifs.tf</code>.</p></li>
</ul></li>
<li><p><strong><em>Optional</em></strong>: Do the same with the <a
href="http://hocomoco11.autosome.ru/">Hocomoco</a> database.</p>
<ul>
<li><p>Note the classification of transcription factors on the home
page. You can browse it to observe TF families, defined based on their
binding domains.</p></li>
<li><p>Find the transcripiton factor of interest and save its
position-counts matrix (PCM) in your work directory.</p></li>
<li><p>With the RSAT tool <em>convert-matrix</em>, convert Hocomoco
motif of your factor to transfac format, and paste it below the Jaspar
motif (make sure that you paste the whole record, including the
<code>//</code> line at the end, which indicates the separation between
two motifs in the same file).</p></li>
<li><p><strong>Tip:</strong> the format Hocomoco is not displayed yet in
the list of input formats for <em>convert-matrix</em>, but it is
actually identical to cluster-buster.</p></li>
</ul></li>
</ul>
</div>
<div id="discovering-motifs-with-rsat-peak-motifs"
class="section level2">
<h2>Discovering motifs with RSAT peak-motifs</h2>
<ul>
<li><p>Open a connection to <a
href="http://rsat.france-bioinformatique.fr/rsat/"><strong>RSAT</strong></a></p></li>
<li><p>Use the tool <strong><a
href="http://rsat.france-bioinformatique.fr/rsat/fetch-sequences_form.php">fetch-sequences</a></strong>
to get the sequences of your peaks.</p>
<ul>
<li><strong>BEWARE:</strong> the peaks of all ChIP-seq datasets were
computed by Ballester’s team based on the <strong>hg38</strong> assembly
of the Human genome. You should thus use this version of the
genome.</li>
</ul></li>
<li><p>Save a copy of the fasta file in your local folder (it might
serve for further analyses).</p></li>
<li><p>At the bottom of the <em>fetch-sequence</em> result page, click
the <strong>peak$motifs button</strong> to send the sequences as input
to <strong>peak-motifs</strong>.</p></li>
<li><p>Set the following parameters for
<strong>peak-mortifs</strong></p>
<ul>
<li>Open the section <em>Compare discovered motifs with databases or
custom reference motifs</em>.</li>
<li>As <em>motif databases</em>, select both <em>Jaspar core
non-redundant vertebrates</em> and <em>Hocomoco Human TFs</em>.</li>
<li>Enter as <em>reference motif</em> the transfac-formatted file with
the JASPAR and Hocomoco motifs for your TF of interest (the file
<code>[YOUR_FACTOR_NAME]_reference-motifs.tf</code> you built in the
previous section).</li>
<li>Enter your email address</li>
<li>Leave all other parameters unchanged an run the analysis.</li>
</ul></li>
<li><p>Interpret the results:</p>
<ul>
<li>Did you obtain significant results? How significant ?</li>
<li>If so, were they characterized by over-representation
(<em>oligo-analysis</em>), positional bias (<em>position-analysis</em>)
or both?</li>
<li>Did some of the discovered motifs match the reference motifs for
your TF in Jaspar and/or Hocomoco?</li>
<li>Did you discover other motifs?</li>
</ul></li>
</ul>
</div>
<div id="negative-control" class="section level2">
<h2>Negative control</h2>
<ul>
<li><p>Use *RSAT random genome fragments** to pick up random regions
having the same sizes as your peaks</p></li>
<li><p>Run the same analyses as above with peak-motifs</p></li>
<li><p>Compare the results of this negative control with those obtained
above with your peaks, and interpret them.</p></li>
</ul>
</div>
<div id="motif-encrihment-analysis" class="section level2">
<h2>Motif encrihment analysis</h2>
<blockquote>
<p>Note : for the 2020 session we had no time to see the theory about
<strong>matrix-quality</strong> and motif enrichment. This section can
thus be skipped for the report.</p>
</blockquote>
<ol style="list-style-type: decimal">
<li><p>On the RSAT server, open the tool
<strong>matrix-quality</strong></p></li>
<li><p>Submit your peak sequences</p></li>
<li><p>Enter the JASPAR and Hocomoco reference motifs</p></li>
<li><p>Run the analyses</p></li>
</ol>
</div>
<div id="motif-discovery-with-meme" class="section level2">
<h2>Motif discovery with MEME</h2>
<p>Analyse the same peak sequences with <a
href="http://meme-suite.org/tools/meme-chip">MEME-ChIP</a>, and convert
the discovered motifs in transfac format with the RSAT tool
<em>convert-matrix</em>.</p>
<p>Also analyse with MEME the random selections of genmic regions
(negative control).</p>
</div>
<div id="matrix-clustering" class="section level2">
<h2>Matrix clustering</h2>
<p>The tool <em>matrix-clustering</em> can be used for different
purposes. Here we will use it to - compare the motifs discovered
respectively by MEME and RSAT peak-motifs in the peak-sequences - merge
the similar motifs in order to obtain a non-redundant collection of
motifs discoverd by the different tools in the peaks - compare this
non-redundant collection of discovered motifs with a motif database in
order to identify the associated TFs.</p>
<ol style="list-style-type: decimal">
<li><p>Run the RSAT tool <em>matrix-clustering</em> to merge and compare
the two collections of motifs discovered by <em>RSAT peak-motifs</em>
and <em>MEME-ChIP</em>, respectively.</p></li>
<li><p>From the <em>matrix-clustering</em> result, retrieve the
<strong>root motifs</strong>, which will provide you with a
non-redundant collection of the motifs resulting from RSAT peak-motifs
and MEME-ChIP.</p></li>
<li><p>Use the RSAT tool <em>compare-matrices</em> to compare these
non-redundant discovered motifs with JASPAR nonredundant
vertebrate.</p></li>
<li><p>Use <em>RSAT matrix-scan</em> to measure the rate of coverage of
the peaks (percentage of peaks with at least one match)</p>
<ul>
<li>for each one of the non-redundant discovered motifs</li>
<li>for each reference motifs</li>
</ul></li>
</ol>
</div>
<div id="interpret-the-results" class="section level2">
<h2>Interpret the results</h2>
<ul>
<li><p>Do the motifs discovered in the peaks include the reference
motifs (those correspinding to the immunoprecipitated factor)?</p></li>
<li><p>Are these motifs reported by all the motif discovery approaches
or only by some of them?</p></li>
<li><p>Do you discover motifs associated with other transcription
factors? If so, are these factors relevant to the ChIP-seq data set
under study (co-factors of the immunoprecipitated factor, factors
involved in this cell/tissue type, …)?</p></li>
<li><p>Which algorithms reported significant motifs in the random
selections of genomic regions? How significant are they compared to
those reported in the actual peaks? How do you interpret these
motifs?</p></li>
</ul>
</div>
<div id="instruction-for-the-report" class="section level2">
<h2>Instruction for the report</h2>
<ol style="list-style-type: decimal">
<li><p>Volume: 4 - 5 pages of text (without counting the figures and the
supp. material)</p></li>
<li><p>Structure: we ask you to return a report structured as a small
research article (introduction, material and methods, results and
discussion, conclusions an perspective).</p></li>
<li><p><strong>Report template.</strong> Please use the template
provided here. Read carefully the instructions.</p>
<ul>
<li><p>Rmd format: <a
href="https://raw.githubusercontent.com/jvanheld/LCG_BEII/gh-pages/practicals/chip-seq_analysis/LASTNAME_Firstname_chip-seq_analysis_report.Rmd">Rmd</a>.</p></li>
<li><p>Word format: <a
href="https://raw.githubusercontent.com/jvanheld/LCG_BEII/gh-pages/practicals/chip-seq_analysis/LASTNAME_Firstname_chip-seq_analysis_report.docx">docx</a>.</p></li>
</ul></li>
</ol>
<p>Please submit both the original file (Rmd) and the report compiled in
Word format (in RStudio, use the command <strong>Knit to Word</strong>),
in order to enable me to annotate it in the margins.</p>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
