---
title: "Practical: analysis of ChIP-seq peaks"
author: "Jacques van Helden"
date: '`r Sys.Date()`'
output:
  html_document:
    code_folding: hide
    self_contained: no
    fig_caption: yes
    highlight: tango
    theme: cerulean
    toc: yes
    toc_depth: 3
    toc_float: yes
  ioslides_presentation:
    slide_level: 2
    self_contained: no
    colortheme: dolphin
    fig_caption: yes
    fig_height: 6
    fig_width: 7
    fonttheme: structurebold
    highlight: tango
    smaller: yes
    toc: yes
    widescreen: yes
  revealjs::revealjs_presentation:
    theme: night
    transition: none
    self_contained: true
    css: ../../slides.css
  slidy_presentation:
    smart: no
    slide_level: 2
    self_contained: yes
    fig_caption: yes
    fig_height: 6
    fig_width: 7
    highlight: tango
    incremental: no
    keep_md: yes
    smaller: yes
    theme: cerulean
    toc: yes
    widescreen: yes
  pdf_document:
    fig_caption: yes
    highlight: zenburn
    toc: yes
    toc_depth: 3
  beamer_presentation:
    colortheme: dolphin
    fig_caption: yes
    fig_height: 6
    fig_width: 7
    fonttheme: structurebold
    highlight: tango
    incremental: no
    keep_tex: no
    slide_level: 2
    theme: Montpellier
    toc: yes
font-import: http://fonts.googleapis.com/css?family=Risque
subtitle: LCG_BEII 2019
font-family: Garamond
transition: linear
---

```{r include=FALSE, echo=FALSE, eval=TRUE}
library(knitr)
library(kableExtra)
# library(formattable)

options(width = 300)
# options(encoding = 'UTF-8')
knitr::opts_chunk$set(
  fig.width = 7, fig.height = 5, 
  fig.path = 'figures/R_intro_',
  fig.align = "center", 
  size = "tiny", 
  echo = TRUE, eval = TRUE, 
  warning = FALSE, message = FALSE, 
  results = FALSE, comment = "")

options(scipen = 12) ## Max number of digits for non-scientific notation
# knitr::asis_output("\\footnotesize")

```


## Introduction

In this practical, we will combine various bioinformatics resources to extract information from ChIP-seq experiments, and evaluate the quality of the peaks.

## Resources


| Resource Name | URL |
|-------------------------|--------------------------------------|
| ReMap | A database of ChIP-seq peaks (<http://remap.cisreg.eu/>) |
| Jaspar | A database of eukaryote TF binding motifs, mainly built from CHIP-seq peaks (<http://jaspar.genereg.net/>) |
| Hocomoco | A database of Human TF binding motifs, mainly built from CHIP-seq peaks (<http://hocomoco11.autosome.ru/>) |
| RSAT| Regulatory Sequence Analysis Tools <http://rsat.eu/> |
| RSAT Metazoa | <http://metazoa.rsat.eu/> |
| UCSC NGS file formats | <https://genome.ucsc.edu/FAQ/FAQformat.html> | 

## Creating a workspace for this practical

- Create a new folder to store the results of this practical (e.g.  `~/LCG_BEII/chip-seq_practical/`)
- Set this folder as your working direct

## Getting peaks from ReMap

ReMap (<http://remap.cisreg.eu/>) is a database of TF binding regions based on an extensive re-analysis of published ChIP-seq data for human transcription factors. ory (`setwd()`)

- Open a connection to ReMap
- Explore the interface
- Choose a transcription factor and a tissue / cell type of interest (for this tutorial I will use Sox2 in Embryonic stem cells, ESC) 
- Click on the **Download by by dataset** tab, and download the peak set corresponding to the tissue / cell type of your choice. 
- Check that the number of peaks in the downloaded file corresponds to the indication on the ReMap page for your factor in the tissue / cell type and in the study identifier (GSE or other) you selected. 
- Check the *name* column in the bed file to make sure that the peaks all belong to the same tissue / cell tupe and study identifier. 

- Compute 

    - the number of peaks
    - the sum of peak lengths
    - the mean peak length
    - the median peak length

**Beware:** the bed format as a somewhat confusiong convention to specify coordinate:

- Coordinates are 0-based (the first position of a sequence is referred to as 0).
- The feature start is (as expected) the first position within the feature, but the "end" is the first position **after the feature**.
- Consequently, the feature length can be computed as $l = end - start$, whereas with other formats it should be $l = end - start = 1$. 

**Tips:**  

- The peak statistics can be computed with a spreadsheet (Excel, LibreOffice Calc), R, or the Unix command line. 

- Click the "code" button to display a bash line to compute peak statistics.

```{bash peak_stats_bash, eval=FALSE, echo=TRUE}
awk '{len=$3 - $2; sum += len; n++; mean = sum / n ; print "n="n"\tsum="sum"\tmean="mean}' \
   practicals/chip-seq_analysis/data/ReMap_GSE69479.SOX2.ESC_peaks.bed \
   | tail -n 1
```

- Click the "code" button to display a chunk of R code to compute peak statistics.

```{r pak_stats_r, eval=TRUE, echo=TRUE}
peakFile <- "data/ReMap_GSE69479.SOX2.ESC_peaks.bed"
peaks <- read.delim(peakFile, header = FALSE)
peaks <- peaks[, 1:4] # Keep the fields used for this analysis
names(peaks) <- c("chr", "start", "end", "name")

peaks$len <- peaks$end - peaks$start
head(peaks) # print the first peaks

sum <- sum(peaks$len)
mean <- mean(peaks$len)
n <- nrow(peaks)

```

In our example (Sox2 peaks in embryonic stem cells from the GEO series GSE69479) we obtain: 

- Number of peaks: $n = `r prettyNum(n, big.mark = ",")`$ 
- Sum of peak lengths: $sum = `r prettyNum(sum, big.mark = ",")`$ 
- Mean peak lengths: $mean = `r round(digits=1, mean)`$ 

## Getting motifs from reference databases

- Open a connection to Jaspar

    - Explore the database functionalities

    - Find the transcription factor binding motifs for your TF of interest. Download the corresponding matrix in your working directory.

- Do the same with the [Hocomoco](http://hocomoco11.autosome.ru/) database. 

    - Note the classification of transcription factors on the home page. You can browse it to observe TF families, defined based on their binding domains. 
    - Find the transcripiton factor of interest and save its binding motif in your work directory. 

- RSAT convert-matrix tools, convert the Jaspar and Hocomoco motifs of your factor to transfac format, and paste them together in a text file (make sure that you paste the whole record, including the `//` line at the end, which indicates the separation between two motifs in the same file). Save this text file as `[YOUR_FACTOR_NAME]_reference-motifs.tf`.

## Discovering motifs with RSAT peak-motifs


- Open a connection to [RSAT Metazoa](http://metazoa.rsat.eu/)

- Use the tool **fetch-sequences** to get the sequences of your peaks. 

    **BEWARE:** the peaks of all ChIP-seq datasets were computed by Ballester's team based on the version hg38 of Human genome

- Send the results to **peak-motifs** 

    - As motif databases, select both Jaspar non-redundant vertebrates and Hocomoco.
    - Enter as reference motif the transfac-formatted file with the JASPAR and Hocomoco motifs for your TF of interest (the file `[YOUR_FACTOR_NAME]_reference-motifs.tf` you built in the previous section). 
    - Leave all other parameters unchanged an run the analysis. 

- Interpret the results: 

    - Did you obtain significant results? How significant ?
    - If so, were they characterized by over-representation (*oligo-analysis*), positional bias (*position-analysis*) or both?
    - Did some of the discovered motifs match the reference motifs for your TF in Jaspar and/or Hocomoco ?
    - Did you discover other motifs
    - 



## Negative control

- Use *RSAT random genome fragments** to pick up random regions having the same sizes as your peaks

- Run the same analyses as above with peak-motifs

- Compare the results of this negative control with those obtained above with your peaks, and interpret them. 



## Motif encrihment analysis

- On the RSAT server, open the tool**matrix-quality**

- Submit your peak sequences

- Enter the JASPAR and Hocomoco reference motifs

- Run the analyses

- Interpret the results

    - Do you observe an enrichment of your peaks for Hocomoco and/or Jaspar reference motifs?
    




## Instruction for the report

1. Volume: 4 - 5 pages of text (without counting the figures and the supp. material)

2. Structure: we ask you to return a report structured as a small research article (introduction, material and methods, results and discussion, conclusions an perspective). 

3. Report template. Please used the template provided here. Read carefully the instructions. 

    - Word format: [docx](LASTNAME_Firstname_chip-seq_analysis_report.docx)
    
    - Rmd format: [Rmd](https://raw.githubusercontent.com/jvanheld/LCG_BEII/gh-pages/practicals/chip-seq_analysis/LASTNAME_Firstname_chip-seq_analysis_report.Rmd). 

***Note that your report should be submitted in Word format, so I can annotate it in the margins. ** If you use Rstudio to write the report, knitr enables you to export in docx format.*